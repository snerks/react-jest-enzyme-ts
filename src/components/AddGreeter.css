.AddGreeter {
  margin: 20px;
  padding: 20px;
  border: 2px solid #00FFD8;
  background: #DDFFEE;
  text-align: center;
}
