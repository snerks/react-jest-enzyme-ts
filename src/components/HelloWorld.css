.HelloWorld {
  border: 2px solid black;
  text-align: center;
  background: #f5f5f5;
  color: #333;
  margin: 20px;
  padding: 20px;
}