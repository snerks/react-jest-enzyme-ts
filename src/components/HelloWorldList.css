.HelloWorldList {
  margin: 20px;
  padding: 20px;
  border: 2px solid #00D8FF;
  background: #DDEEFF;
}