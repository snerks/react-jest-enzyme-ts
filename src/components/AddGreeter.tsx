import * as React from 'react';
import './AddGreeter.css';

export interface AddGreeterProps {
    addGreeting: Function;
}

export interface AddGreeterState {
    greetingName: string;
}

class AddGreeter extends React.Component<AddGreeterProps, AddGreeterState> {
    constructor(props: AddGreeterProps) {
        super(props);

        this.state = { greetingName: '' };
    }

    // handleUpdate(event: React.ChangeEvent<HTMLInputElement>) {
    //     this.setState({ greetingName: event!.target!.value });
    // }

    // tslint:disable-next-line:no-any
    handleUpdate(event: any) {
        this.setState({ greetingName: event!.target!.value });
    }

    addGreeting() {
        this.props.addGreeting(this.state.greetingName);
        this.setState({ greetingName: '' });
    }

    render() {
        return (
            <div className="AddGreeter">
                <input
                    type="text"
                    value={this.state.greetingName}
                    onChange={(e) => this.handleUpdate(e)}
                />
                &nbsp;&nbsp;
                <button onClick={() => this.addGreeting()}>Add</button>
            </div>
        );
    }
}

export default AddGreeter;
