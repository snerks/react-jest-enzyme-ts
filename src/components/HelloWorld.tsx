import * as React from 'react';
import './HelloWorld.css';

export interface HelloWorldProps {
    name: string;
    removeGreeting: Function;
}

export interface HelloWorldState {
    greeting: string;
}

class HelloWorld extends React.Component<HelloWorldProps, HelloWorldState> {

    constructor(props: HelloWorldProps) {
        super();

        this.state = { greeting: 'Hello' };
    }

    frenchify() {
        this.setState({ greeting: 'Bonjour' });
    }

    removeGreeting() {
        this.props.removeGreeting(this.props.name);
    }

    render() {
        return (
            <div className="HelloWorld">
                {this.state.greeting} {this.props.name}!
                <br />
                <button className="frenchify" onClick={() => this.frenchify()}>Frenchify!</button>
                <br />
                <button className="remove" onClick={() => this.removeGreeting()}>Remove Me!</button>
            </div>
        );
    }
}

export default HelloWorld;
