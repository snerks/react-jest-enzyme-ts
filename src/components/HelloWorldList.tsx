import * as React from 'react';
import HelloWorld from './HelloWorld';
import AddGreeter from './AddGreeter';

export interface HelloWorldListProps {
}

export interface HelloWorldListState {
    greetings: string[];
}

class HelloWorldList extends React.Component<HelloWorldListProps, HelloWorldListState> {
    constructor(props: HelloWorldListProps) {
        super(props);
        this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
    }

    addGreeting(newName: string) {
        this.setState({ greetings: [...this.state.greetings, newName] });
    }

    removeGreeting(removeName: string) {
        const filteredGreetings = this.state.greetings.filter(name => {
            return name !== removeName;
        });

        this.setState({ greetings: filteredGreetings });
    }

    renderGreetings() {
        return this.state.greetings.map(name => (
            <HelloWorld
                key={name}
                name={name}
                removeGreeting={(removeName: string) => this.removeGreeting(removeName)}
            />
        ));
    }

    render() {
        return (
            <div className="HelloWorldList">
                <AddGreeter addGreeting={(newName: string) => this.addGreeting(newName)} />
                {this.renderGreetings()}
            </div>
        );
    }
}

export default HelloWorldList;
