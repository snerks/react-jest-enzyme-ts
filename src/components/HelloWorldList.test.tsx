import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import AddGreeter from './AddGreeter';
import HelloWorld from './HelloWorld';

describe('HelloWorldList', () => {
    const _component = shallow(
        <HelloWorldList />
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWorldList />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains an AddGreeter subcomponent', () => {
        expect(_component.find(AddGreeter)).toHaveLength(1);
    });

    it('contains the same number of HelloWorld components as greetings', () => {
        const helloWorldCount = _component.find(HelloWorld).length;
        const greetingCount = (_component.state('greetings') as string[]).length;
        expect(helloWorldCount).toEqual(greetingCount);
    });

    it('adds another greeting when the add greeting function is called', () => {
        const beforeHelloWorldCount = _component.find(HelloWorld).length;

        (_component.instance() as HelloWorldList).addGreeting('Sample');

        const afterHelloWorldCount = _component.find(HelloWorld).length;
        expect(afterHelloWorldCount).toBeGreaterThan(beforeHelloWorldCount);
    });

    it('removes a greeting from the list when the remove greeting function is called', () => {
        const beforeHelloWorldCount = _component.find(HelloWorld).length;

        const removeGreetingName = (_component.state('greetings') as string[])[0];
        (_component.instance() as HelloWorldList).removeGreeting(removeGreetingName);

        const afterHelloWorldCount = _component.find(HelloWorld).length;
        expect(afterHelloWorldCount).toBeLessThan(beforeHelloWorldCount);
    });
});