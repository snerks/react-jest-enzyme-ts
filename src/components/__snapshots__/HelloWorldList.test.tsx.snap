exports[`HelloWorldList renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList">
  <div
    className="AddGreeter">
    <input
      onChange={[Function]}
      type="text"
      value="" />
      
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Jim
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Sally
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Bender
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
</div>
`;
